06_puppy/nickolee/types.go

```diff
@@ -0,0 +1,31 @@
+package puppystorer
+
+import "sync"
+
+type Puppy struct {
+	ID     int
+	Breed  string
+	Colour string
+	Value  float64
+}
+
+// Storer defines standard CRUD operations for Pets
+type Storer interface {
+	CreatePuppy(puppy *Puppy) int // takes an address as an argument which makes sense since you are modifying that object
+	ReadPuppy(id int) (*Puppy, error)
+	UpdatePuppy(id int, puppy *Puppy) error
+	DeletePuppy(id int) (bool, error)
+}
+
+// type MapStore map[int]Puppy
+// This will serve as our in-memory DB
+type MapStore struct {
+	store   map[int]Puppy
+	counter int
```

Comment:
would call this nextID (that's how it's used) and increment it after asignement.

Reply:
Thanks for the tip. Have renamed.